### ISSUE TITLE
Regression: Erratic left/right intra-day hours navigation: Affects partial hours and DST support

### ISSUE THREAD

**Issue by jnd-au** [oz/tz#60]
Unreleased changes in HEAD (e.g. cb3ba9f after release v0.7.0) cause errors and instability in `tz` when the user is in, or is viewing, timezones with fractional-hour offsets (e.g. 30 or 45 minutes different than UTC), and days when daylight saving time (DST) starts and ends, and left/right keyboard navigation. Furthermore, keyboard navigation gets stuck on DST transition days (it’s broken)!

**Comment by wyne** [comment]
I haven't noticed this. Do you have an example set of timezones to reproduce?

**Comment by jnd-au** [comment]
Sure @wyne, including examples that are shown, tested, and resolved in PR #62. Some quick examples are India and Australia:
- Hour navigation left gets stuck on hour 2 of DST transition, stops moving: `./tz.0bdd6e2 -when 1712422800 Australia/Sydney` (pressing `h` moves to hour 2 then gets stuck).
- Hour navigation left/right for fractional timezones causes the hour to change irreversibly (cannot get back to starting time or vertical alignment): `./tz.cb3ba9f -when 1727749800 Asia/Calcutta,India`

```
# See original time 8:00 AM:

  🕗 (IST) India                                                                         8:00AM, Tue Oct 01, 2024
  20  21  22  23   0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  

# Press l (forward hour)

  🕗 (IST) India                                                                         8:30AM, Tue Oct 01, 2024
  19  20  21  22  23   0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  

# See above that the hours have shifted, and the time is shown as 8:30 AM instead of 9:00 AM.
# Now try to go back to the original time: Press h

  🕖 (IST) India                                                                         7:30AM, Tue Oct 01, 2024
  19  20  21  22  23   0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  

# See hours have not reverted to the start, and the time is shown as 7:30 AM instead of the original 8:00 AM
```
